pkgs/by-name/us/usque/package.nix

```
@@ -0,0 +1,51 @@
+{
+  fetchFromGitHub,
+  buildGoModule,
+  lib,
+  installShellFiles,
+  versionCheckHook,
+}:
+buildGoModule rec {
+  pname = "usque";
+  version = "1.0.2";
+  src = fetchFromGitHub {
+    owner = "Diniboy1123";
+    repo = "usque";
+    tag = "v${version}";
+    hash = "sha256-0nIKBLjuh0chycpCuqG31/TGe8oTzdENS8sTFYHA3I8=";
+  };
+
+  vendorHash = "sha256-njkwrzw/8m4Y1l8aGxaK+JrYbKo/7pCT/ck0bbdaNbU=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/Diniboy1123/usque/cmd.version=${version}"
+  ];
+
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+  postInstall = ''
+    installShellCompletion --cmd usque \
+      --bash <($out/bin/usque completion bash) \
+      --fish <($out/bin/usque completion fish) \
+      --zsh <($out/bin/usque completion zsh)
+  '';
```

would be better to use emulator here (it's what's done in [the example for installShellCompletion](https://nixos.org/manual/nixpkgs/unstable/))
```suggestion
  postInstall = lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) (
    let
      emulator = stdenv.hostPlatform.emulator buildPackages;
    in
    ''
      installShellCompletion --cmd usque \
        --bash <(${emulator} $out/bin/usque completion bash) \
        --fish <(${emulator} $out/bin/usque completion fish) \
        --zsh <(${emulator} $out/bin/usque completion zsh)
    ''
  );
```

---

Guess we can do a treewide change for all the packages that do not have this guard.